source/simulator/parameters.cc

```diff
@@ -858,6 +888,24 @@ namespace aspect
                    ExcMessage("Invalid input parameter file: Too many entries in List of normalized fields"));
     }
     prm.leave_subsection ();
+
+
+    // now also get the parameters related to material model averaging
+    prm.enter_subsection ("Material model");
+    {
+      material_averaging_for_linear_systems
+        = MaterialModel::MaterialAveraging::parse_averaging_operation_name
+          (prm.get ("Material averaging for linear systems"));
+      material_averaging_for_preconditioners
+        = MaterialModel::MaterialAveraging::parse_averaging_operation_name
+          (prm.get ("Material averaging for preconditioners"));
+
+      // implement the special rule that if the linear systems use some
+      // averaging, then the preconditioner needs to use the same
+      if (material_averaging_for_linear_systems != MaterialModel::MaterialAveraging::none)
```

Comment:
Good question. I meant to still add this to the manual, but will have to work some for that.

The idea is that if you do average, then the same averaging should also be used for the preconditioners (because averaging is always better for the solver, there is no point in not averaging the preconditioner if you average the matrix). But you may choose to solve the original system and in that case it may still be better to average the preconditioner -- it'll be the preconditioner for the wrong system, but it may be faster than a non-averaged preconditioner.

Will try out and report back.